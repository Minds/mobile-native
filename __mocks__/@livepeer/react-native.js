// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
