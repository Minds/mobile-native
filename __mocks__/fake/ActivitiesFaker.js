function load(count, container) {
  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid: code,
      custom_data: false,
      custom_type: false,
      rowKey: 'something' + i,
      description: 'Congratulations! ',
      edited: '',
      guid: code,
      mature: false,
      time_created: '1522036284',
      ownerObj: {
        guid: '824853017709780997',
        type: 'user',
        subtype: false,
        time_created: '1522036284',
      },
      message: 'Message',
      title: 'TITLE',
      owner_guid: '824853017709780997',
      parent_guid: '838106762591510528',
      perma_url: false,
      thumbnail_src: false,
      type: 'activity',
      wire_totals: {
        tokens: 1000000000000000000,
      },
      containerObj: container,
    };
  });

  return {
    activities: activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load };
}

/**
 * Generate a single fake entity
 * @param {object} containerObj optional entity container
 * @returns
 */
export function fakeOne(containerObj) {
  const resp = load(1, containerObj);
  return resp.activities[0];
}
