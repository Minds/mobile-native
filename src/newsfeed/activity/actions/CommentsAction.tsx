import React, { Component, useCallback, useEffect } from 'react';

import { observer } from 'mobx-react';

import { TouchableOpacity } from 'react-native';

import Icon from 'react-native-vector-icons/MaterialIcons';

import Counter from './Counter';
import withPreventDoubleTap from '../../../common/components/PreventDoubleTap';
import ThemedStyles from '../../../styles/ThemedStyles';
import type ActivityModel from '../../../newsfeed/ActivityModel';
import type BlogModel from '../../../blogs/BlogModel';
import { useRoute } from '@react-navigation/native';
import { ActivityRouteProp } from '../../ActivityScreen';

// prevent double tap in touchable
const TouchableOpacityCustom = withPreventDoubleTap(TouchableOpacity);

type PropsType = {
  entity: ActivityModel | BlogModel;
  testID?: string;
  navigation: any;
  onPressComment?: () => void;
};

/**
 * Comments Action Component
 */
const CommentsAction = observer((props: PropsType) => {
  const theme = ThemedStyles.style;
  const icon = props.entity.allow_comments
    ? 'chat-bubble'
    : 'speaker-notes-off';

  const route: ActivityRouteProp = useRoute();

  const openComments = useCallback(() => {
    if (props.onPressComment) {
      props.onPressComment();
      return;
    }
    const cantOpen =
      !props.entity.allow_comments && props.entity['comments:count'] === 0;

    if ((route && route.name === 'Activity') || cantOpen) {
      return;
    }
    if (props.entity.subtype && props.entity.subtype === 'blog') {
      props.navigation.push('BlogView', {
        blog: props.entity,
        scrollToBottom: true,
      });
    } else {
      props.navigation.push('Activity', {
        entity: props.entity,
        scrollToBottom: true,
      });
    }
  }, [props, route]);

  useEffect(() => {
    if (route && (route.params?.focusedUrn || route.params?.scrollToBottom)) {
      setTimeout(() => {
        openComments();
      }, 100);
    }
  }, [openComments, route]);

  return (
    <TouchableOpacityCustom
      style={[
        theme.rowJustifyCenter,
        theme.paddingHorizontal3x,
        theme.paddingVertical4x,
        theme.alignCenter,
      ]}
      onPress={openComments}
      testID={props.testID}>
      <Icon
        style={[theme.colorIcon, theme.marginRight]}
        name={icon}
        size={21}
      />
      <Counter count={props.entity['comments:count']} />
    </TouchableOpacityCustom>
  );
});

export default CommentsAction;
