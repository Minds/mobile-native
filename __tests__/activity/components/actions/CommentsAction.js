import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../../__mocks__/fake/ActivitiesFaker';

import CommentsAction from '../../../../src/newsfeed/activity/actions/CommentsAction';

describe('Comment action component', () => {
  let screen, navigatorStore, navigation;
  beforeEach(() => {
    navigation = {
      push: jest.fn(),
      state: { routeName: 'some' },
      dangerouslyGetState: jest.fn(),
    };
    let activityResponse = activitiesServiceFaker().load(1);
    screen = shallow(
      <CommentsAction
        entity={activityResponse.activities[0]}
        navigation={navigation}
      />,
    );

    //jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a comment button', async () => {
    screen.update();

    expect(screen.find('preventDoubleTap(TouchableOpacity)')).toHaveLength(1);
  });

  it('should navigate a thumb on press ', async () => {
    navigation.dangerouslyGetState.mockReturnValue({ routes: null });

    screen.update();
    let touchables = screen.find('preventDoubleTap(TouchableOpacity)');
    touchables.at(0).props().onPress();
    expect(navigation.push).toHaveBeenCalled();
    expect(screen.find('preventDoubleTap(TouchableOpacity)')).toHaveLength(1);
  });
});
