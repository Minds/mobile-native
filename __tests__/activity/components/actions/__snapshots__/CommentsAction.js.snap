// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment action component renders correctly 1`] = `
<preventDoubleTap(TouchableOpacity)
  onPress={[Function]}
  style={
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <Icon
    allowFontScaling={false}
    name="speaker-notes-off"
    size={21}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <Counter
    size={15}
  />
</preventDoubleTap(TouchableOpacity)>
`;
